internal/services/managedredis/managed_redis_resource.go

```diff
@@ -680,6 +685,22 @@ func (r ManagedRedisResource) CustomizeDiff() sdk.ResourceFunc {
 				}
 			}
 
+			if metadata.ResourceDiff.HasChanges("sku_name") {
+				if resId := metadata.ResourceDiff.Id(); resId != "" {
+					clusterId, err := redisenterprise.ParseRedisEnterpriseID(resId)
+					if err != nil {
+						return err
+					}
+					clusterClient := metadata.Client.ManagedRedis.Client
+					warning, err := isSkuAllowedForScaling(ctx, clusterClient, clusterId, model.SkuName)
+					if err != nil {
+						metadata.ResourceDiff.ForceNew("sku_name")
+					} else if warning != "" {
+						metadata.Logger.Warnf(warning)
+					}
```

Comment:
Please keep the error / warning handling internal to isSkuAllowedForScaling function, this logic should be simplified like this:

```suggestion
					if !isSkuAllowedForScaling(ctx, clusterClient, clusterId, model.SkuName) {
						metadata.ResourceDiff.ForceNew("sku_name")
					}
```

Reply:
My first impulse was to keep only the decision on the function, but since that is only used once i guess that makes no difference. Is Done.